core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -304,8 +316,8 @@ private SearcherManager createSearcherManager() throws EngineException {
                 throw new EngineCreationFailureException(shardId, "failed to open reader on writer", e);
             }
         } finally {
-            if (success == false) { // release everything we created on a failure
-                IOUtils.closeWhileHandlingException(searcherManager, indexWriter);
+            if (searcherManager == null) { // release everything we created on a failure
```

Comment:
I think this may a leak an unclosed searchManager if the second one runs into trouble (than only the second one will be closed). Maybe we should deal with all of that the constructor? we already have a try catch there?